docs/defaults-detailed.rst

```diff
@@ -9,169 +11,640 @@ them.
-  configured as ``static``, the first IPv4 address will be the main one, the rest
-  will be added with numbered labels. When the interface is configured to get its
-  IP address via DHCP, all IPv4 addresses will be added with numbered labels.
-
-``gateway``
-  IP address of the gateway configured for this interface; only one gateway can
-  be configured at the moment.
-
-``label``
-  Specify a string which will be added to the interface name as a label.
-  Labeled interface configuration is stored in separate configuration files in
-  :file:`/etc/network/interfaces.d/`.
+  Optional, boolean, string or YAML list. If set to ``False``, this option is
+  disabled. If ``True``, the hotplug subsystem can bring this interface up or
+  down when the hotplug event is detected. You can also specify a list of
+  specific conditions at which the interface is brought up, currently
+  recognized conditions are:
+
+  - ``auto``: bring the interface up at boot time by the ``networking``
+    service. This might not be what you want on newer systems.
+
+  - ``boot``: bring the interface up at boot time by ``ifup@.service``
+    :command:`systemd` unit. This will put any processes related to a given
+    interface in their separate cgroup, which allows for better control over
+    the network interface. This is a custom implementation of the ``auto``
+    mechanism managed by this Ansible role.
+
+  - ``hotplug``: bring the interface up/down at hotplug events. This condition
+    is required to be present for the ``ifup@.service`` :command:`systemd` unit
+    to work properly.
+
+IPv4 and IPv6 configuration parameters
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``inet``
+  Optional. IPv4 configuration method used by a given interface. There are many
+  configuration methods described in the :manpage:`interfaces(5)` manual page, most
+  commonly used are: ``manual``, ``dhcp``, ``static``. If you set this
+  parameter to ``False``, the IPv4 configuration will be disabled.
+
+``inet6``
+  Optional. IPv6 configuration method used by a given interface. There are many
+  configuration methods described in the :manpage:`interfaces(5)` manual page, most
+  commonly used are: ``auto``, ``manual``, ``dhcp``, ``static``, ``v4tunnel``,
+  ``6to4``. If you set this parameter to ``False``, the IPv4 configuration will
+  be disabled.
+
+``address`` or ``addresses``
+  Optional. A string or an YAML list of IPv4 and/or IPv6 addresses to set on
+  a given network interface, in the form of ``ipaddress/prefix`` or CIDR.
+  Remember that you need to specify the host IP address and not the network;
+  the ``192.0.2.1/24`` is the correct notation, and ``192.0.2.0/24`` is
+  incorrect.
+
+``gateway`` or ``gateway4``
+  Optional. Specify the IPv4 address of the network gateway to which outgoing
+  packets will be directed.
+
+``gateway6``
+  Optional. Specify the IPv6 address of the network gateway to which outgoing
+  packets will be directed.
```

Comment:
Hmmm, I guess that would make it similar to how `address` and `addresses` work. Agreed.